app/models/name.rb

```diff
@@ -439,7 +439,14 @@ def real_search_name
   def self.display_to_real_text(name)
     name.display_name.gsub(/ ^\*?\*?__ | __\*?\*?[^_\*]*$ /x, "").
       gsub(/__\*?\*? [^_\*]* \s (#{ANY_NAME_ABBR}) \s \*?\*?__/x, ' \1 ').
-      gsub(/__\*?\*? [^_\*]* \*?\*?__/x, " ") # (this part should be unnecessary)
+      gsub(/__\*?\*? [^_\*]* \*?\*?__/x, " "). # (this part should be unnecessary)
+      # Because "group" was removed by the 1st gsub above,
+      # tack it back on (if it was part of display_name)
+      concat(group_suffix(name))
```

Comment:
This line is necessary because the first `gsub` removes everything in the name after the taxon, including “group” (and author). For “group” names, `display_name` now includes “group” and so should `real_text_name`.
There's probably a way to revise the `gsub` so it does not remove "group".  But I couldn't figure it out, so it's probably convoluted.